### ISSUE TITLE
Dialog is missing ARIA attributes

### ISSUE THREAD

**Issue by trilodge** [SAP/fundamental-styles#848]
**Describe the bug**
The markup of the dialog component is missing some crucial aria attributes regarding accessibility.

It would be better if the dialog follows the w3c wai aria recommendation like given here:
https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

This issue will also be mirrored to fundamental ngx: https://github.com/SAP/fundamental-ngx/issues/2299 because the essential attributes are missing there too.

Attributes needed:

role="dialog"
aria-labeledby="xxxxxx" -> linked to the headline of the dialog

> Is this issue related to a specific component?
Dialog
> What version of the Fundamental Library Styles are you using?
0.8.x ++
> What front-end framework are you implementing in, e.g., Angular, React, etc.?
angular

**Desktop (please complete the following information):**
 - all browsers

**Additional context**
https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html
